pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -102,6 +111,43 @@ func (c *StaticResourceController) WithIgnoreNotFoundOnCreate() *StaticResourceC
 	return c
 }
 
+func shouldDelete(createConditionalFunc, deleteConditionalFunc resourceapply.ConditionalFunction) bool {
+	shouldDelete := false
+	if createConditionalFunc == nil && deleteConditionalFunc == nil {
+		shouldDelete = false
+	} else {
+		if createConditionalFunc != nil && createConditionalFunc() {
```

Comment:
delete should only be controlled by the deletionfunc.  Create should only be controlled by the creation func.

if the create and delete func both report true, an error should be reported (do this in sync)